src/compiler/crystal/command.cr

```diff
@@ -446,8 +446,11 @@ class Crystal::Command
       end
 
       unless no_codegen
-        opts.on("--release", "Compile in release mode") do
-          compiler.release = true
+        opts.on("--release", "Compile in release mode (-O3 --single-module)") do
+          compiler.release!
+        end
+        opts.on("-O LEVEL", "LLVM optimization level") do |level|
```

Comment:
**suggestion:** Make this option more general. It does not only affect LLVM optimization level. 

The inline threshhold is also affected. And there may be further enhancements outside of LLVM codegen.

I like the internal term "mode" and would use that in the CLI docs as well.
Stating at least which modes are available should be a good hint for users.

```suggestion
        opts.on("-O MODE", "Optimization mode: 0 (default), 1, 2, 3") do |level|
```